app/tests/spec/views/settings.js

```diff
@@ -521,6 +520,52 @@ function (chai, $, sinon, Cocktail, View, BaseView, SubPanels,
         });
       });
 
+      // the following three tests
+      // confirm that the dom displays correctly
+      // when the _swapDisplayName method is called
+      describe('user has no display name', function () {
+        it('should show email address in header', function () {
+
+          return view.render()
+            .then(function () {
+              view._swapDisplayName();
+
+              assert.equal(view.$('.card-header').text(), 'a@a.com');
+              assert.equal(view.$('.card-subheader').text(), '');
+            });
+        });
+      });
+
+      describe('with display name removed', function () {
+        it('should place the email in the header', function () {
+          account.set('displayName', 'test user');
+
+          return view.render()
+            .then(function () {
+              account.unset('displayName');
+
+              view._swapDisplayName();
+
+              assert.equal(view.$('.card-header').text(), 'a@a.com');
+              assert.equal(view.$('.card-subheader').text(), '');
+            });
+        });
+      });
+
+      describe('with display name added', function () {
+        it('should place the display name in the header and email in the subheader', function () {
+          account.set('displayName', 'test user');
+
+
+          return view.render()
+            .then(function () {
+              view._swapDisplayName();
+
+              assert.equal(view.$('.card-header').text(), 'test user');
+              assert.equal(view.$('.card-subheader').text(), 'a@a.com');
+            });
+        });
+      });
```

Comment:
As noted [above](https://github.com/mozilla/fxa-content-server/pull/3089/files#r40900440), one more test is needed that starts with the email address in the header, then adds a display name after the initial render. That should hit the uncovered code.